src/domain/wallet-invoices/wallet-invoice-factory.ts

```diff
@@ -0,0 +1,30 @@
+export const WalletInvoiceFactory = (walletId: WalletId): IWalletInvoiceFactory => {
+  const create = ({ registeredInvoice }): WalletInvoice => {
+    return _baseCreate({ registeredInvoice, selfGenerated: true })
```

Comment:
I find the indirection here overkill. Would prefer to see the 2 functions `create` + `createForRecipient` return their own `WalletInvoice` inline. Makes it easier to read on first pass without having to understand the `_baseCreate`.